src/portfolios/portfolios.service.ts

```diff
@@ -78,4 +79,20 @@ export class PortfoliosService {
       transactionBaseDto
     );
   }
+
+  public async updatePortfolioName(
+    portfolioParams: PortfolioDto,
+    params: ModifyPortfolioDto
+  ): ServiceReturn<NumberedPortfolio> {
+    const { did, id } = portfolioParams;
+    const { base, args } = extractTxBase(params);
+
+    const portfolio = (await this.findOne(did, id)) as NumberedPortfolio;
+
+    if (!portfolio.modifyName) {
+      throw new BadRequestException('Default portfolio name cannot be modified');
```

Comment:
```suggestion
      throw new AppValidationError('Default portfolio name cannot be modified');
```
There is a filter that will convert it for you so the behavior will be the same when used with the HTTP controller.

With something like REPL mode an HTTP error doesn't always make sense, which is what motivated me to make the new error type. 

The illustrative example is exposing the services over RPC.  It would be a leaky abstraction to have a different set of endpoints coupled to HTTP details.

Reply:
I admit its bit nitty, since I think any one using a REPL knows what HTTP codes are, but I think the cleaner abstraction is worth it.